src/TestStack.White/InputDevices/IKeyboard.cs

```diff
@@ -1,13 +1,81 @@
+using TestStack.White.UIItems.Actions;
 using TestStack.White.WindowsAPI;
 
 namespace TestStack.White.InputDevices
 {
     public interface IKeyboard
     {
+        /// <summary>
+        /// Enter a string
+        /// </summary>
+        /// <param name="keysToType">String to enter</param>
         void Enter(string keysToType);
+
+        /// <summary>
+        /// Send a strig to a specific <see cref="IActionListener"/>
+        /// </summary>
+        /// <param name="keysToType">String to send</param>
+        /// <param name="actionListener"><see cref="IActionListener"/> to receive the string</param>
+        void Send(string keysToType, IActionListener actionListener);
```

Comment:
As far as I know, the main reason for the IActionListener is for recording what is happening.

If the user passes a custom IActionListener to the UIItem then it will be used with the keyboard decorator. OR we provide an overload to Keyboard and it calls if it's required. Both of these remove the duplication of members on `IAttachedKeyboard` and `IKeyboard` and also remove all the methods taking an IActionListener?